File: Array-Level-0/maxConsecutiveOnes/maxConsecutiveOnes.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std; 

int findMaxConsecutiveOnes(vector<int> arr) {
    int currentOnes = 0; 
    int maxOnes = 0; 
    
    for(int i = 0;i<arr.size();i++) {
        if(arr[i] == 1) {
            currentOnes++; 
        } else {
            currentOnes = 0;
        }
        
        maxOnes = max(currentOnes , maxOnes);
    }
    return maxOnes;
}

int main() {
    vector<int> arr = {1,1,0,1,1,1};  //3
    int maxOnes = findMaxConsecutiveOnes(arr);
    cout << "The max ones in the array are : " << maxOnes << endl; 
}

File: 01 - ArrayLevel-EasyMedium/maxConsecutiveOnes/maxConsecutiveOnes.js
function maxConsecutiveOnes(arr) {
  let currentOnes = 0;
  let maxOnes = 0;
  for (let i = 0; i < arr.length;i++) {
    if(arr[i] == 1) {
      currentOnes++;
    } else {
      currentOnes = 0; 
    }
    
    maxOnes = Math.max(currentOnes, maxOnes);
  }
  return maxOnes; 
}

let array = [1,0,1,1,1,1,1,1,0,1]
let maxOnesInArray = maxConsecutiveOnes(array);
console.log(`The max ones in the array ${array} are : ${maxOnesInArray}`);


